node-runtime/src/encode-decode.ts

```diff
@@ -20,6 +20,17 @@ const simpleTypes = [
     ...simpleStringTypes,
 ];
 
+declare global {
+    interface BigInt {
+        toJSON(): string;
+    }
+}
+
+// eslint-disable-next-line no-extend-native
+BigInt.prototype.toJSON = function toJSON() {
+    return this.toString();
+};
+
```

Comment:
The encode function below already converts to string (line 170). This shouldn't be needed.

This was the real fix:

    if (!(typeof value === "bigint"))